test/uniport-tests.js

```diff
@@ -0,0 +1,502 @@
+    {
+      skip_if_not_match:configurations.skip_tests_not_matching,
+      skip_if_match:configurations.skip_tests_matching
+    },
+    global.test );
+
+var define_suite = testutils.definer_matching(
+    {
+      skip_if_not_match:configurations.skip_ports_not_matching,
+      skip_if_match:configurations.skip_ports_matching
+    },
+    global.suite );
+
+mark_redundant_tests = function(normalizer) {
+  var tested = [];
+
+  return function(test_name, func) {
+    var redundant;
+    var normed = normalizer ? normalizer(test_name) : test_name;
+
+    if (!(~tested.indexOf(normed))) {
+      redundant = false;
+      tested.push(normed);
+    } else {
+      redundant = true;
+      test_name += (' (redundant)')
+    }
+    if (!redundant || configurations.define_redundant_tests) {
+      define_test(test_name, func);
+    }
+  }
+}
+
+var add_credentials_to_request = function(message, credentials, wrong_pass,
+                                                                wrong_user)
+{
+  message.admin_user = credentials.admin_user;
+  message.admin_password = credentials.admin_password;
+
+  if (wrong_pass) {
+    if (wrong_pass === 'object') {
+      message.admin_password = {admin_password: message.admin_password};
+    } else {
+       message.admin_password += '_';
+    }
+  };
+
+  if (wrong_user) {
+    message.admin_user += '_';
+  };
+}
+
+/* ---------------------------------- TESTS --------------------------------- */
+
+var test_websocket_admin_command = function (test_decl, protocol, conf, done)
+{
+  var should_work = test_decl.should_work;
+  var should_not_work = !should_work;
+  var require_pass = conf.admin_password != null;
+  var send_credentials = test_decl.send_credentials;
+  var wrong_pass = test_decl.wrong_pass;
+  var wrong_user = test_decl.wrong_user;
+
+  var config = {
+    'websocket_ip':  "127.0.0.1",
+    'websocket_port': conf.port,
+    'websocket_ssl': protocol === 'wss',
+    'trace' : configurations.remote_trace
+  };
+
+  var remote = Remote.from_config(config);
+
+  if (require_pass) {
+    remote.on('prepare_subscribe', function(request){
+      if (require_pass && configurations.add_credentials_to_ws_subscribe) {
+        add_credentials_to_request(request.message, conf);
+      };
+      request.once('error', function(e, m){
+        if (!configurations.add_credentials_to_ws_subscribe) {
+          assert.notEqual(e.remote.error,
+                          'forbidden',
+                          'Need credentials for non admin request (subscribe)')
+        };
+      })
+    });
+  };
+
+  remote.once('connected', function() {
+    var b4_ledger_accept = remote._ledger_current_index;
+    var request = new Request(remote, 'ledger_accept');
+
+    if (send_credentials) {
+      add_credentials_to_request(request.message, conf, wrong_pass, wrong_user);
+    }
+
+    request.request(function(e, m){
+      remote.connect(false);
+      var M = function(message) {
```

Comment:
You can't come up with a better name than `M`?  :-P

(Also, use the standard `function betterNameHere(message) {` form...)